addons/pinniped/config-controller/controllers/controllers.go

```diff
@@ -197,35 +209,39 @@ func (c *PinnipedController) reconcileAddonSecret(ctx context.Context, cluster *
 
 	log.V(1).Info("retrieved data from pinniped-info configmap", "supervisorAddress", supervisorAddress, "supervisorCABundle", supervisorCABundle)
 
-	pinnipedAddonSecret := &corev1.Secret{
+	pinnipedSecret := &corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
-			Namespace: cluster.Namespace,
-			Name:      fmt.Sprintf("%s-pinniped-addon", cluster.Name),
-			Labels: map[string]string{
-				constants.TKGAddonLabel:       constants.PinnipedAddonLabel,
-				constants.TKGClusterNameLabel: cluster.Name,
-			},
-			Annotations: map[string]string{
-				constants.TKGAddonTypeAnnotation: constants.PinnipedAddonTypeAnnotation,
-			},
+			Namespace: secretNamespace,
+			Name:      secretName,
 		},
 	}
-	// TODO: remove this
-	pinnipedAddonSecret.Type = constants.TKGAddonType
-	pinnipedAddonSecret.Data = make(map[string][]byte)
 
-	pinnipedAddonSecretMutateFn := func() error {
+	pinnipedSecretMutateFn := func() error {
```

Comment:
Kind of inclined to extract this inline func out into a separate named func since it is big enough to be "significant".

Reply:
(also ok to leave as a refactor TODO)